src/content-handlers/iiif/modules/uv-openseadragoncenterpanel-module/OpenSeadragonCenterPanel.ts

```diff
@@ -925,6 +940,60 @@ export class OpenSeadragonCenterPanel extends CenterPanel<
     this.isFirstLoad = false;
   }
 
+  private resizeNavigatorForContinuous(): void {
+    if (!this.viewer || !this.viewer.navigator) return;
+
+    const homeBounds = this.viewer.world.getHomeBounds();
+    const contentAspectRatio = homeBounds.width / homeBounds.height;
+
+    const viewportWidth = this.$viewer.width();
+    const viewportHeight = this.$viewer.height();
+    const maxNavigatorWidth = 100;
+    const maxNavigatorHeight = 100;
+    const minVerticalNavigatorWidth = 60;
+    const minHorizontalNavigatorHeight = 40;
+
+    let navigatorWidth: number;
+    let navigatorHeight: number;
+
+    if (this.extension.helper.isVerticallyAligned()) {
+      navigatorHeight = viewportHeight - this.$zoomInButton.height() - 4;
+      navigatorWidth = navigatorHeight * contentAspectRatio;
+
+      // Enforce max width
+      if (navigatorWidth > maxNavigatorWidth) {
+        navigatorWidth = maxNavigatorWidth;
+      }
+
+      // Enforce min width
+      if (navigatorWidth < minVerticalNavigatorWidth) {
+        navigatorWidth = minVerticalNavigatorWidth;
+      }
+    } else {
+      navigatorWidth = viewportWidth;
+      navigatorHeight = navigatorWidth / contentAspectRatio;
+
+      // Enforce max height
+      if (navigatorHeight > maxNavigatorHeight) {
+        navigatorHeight = maxNavigatorHeight;
+      }
+
+      // Enforce min height
+      if (navigatorHeight < minHorizontalNavigatorHeight) {
+        navigatorHeight = minHorizontalNavigatorHeight;
+      }
+    }
+
+    const navigatorElement = this.viewer.navigator.element;
+    navigatorElement.style.width = `${navigatorWidth}px`;
+    navigatorElement.style.height = `${navigatorHeight}px`;
+
+    //add css to fit with title here?
```

Comment:
Making this a TODO so we don't lose track of it.

```suggestion
    // TODO: add css to fit with title here?
```